versioned_docs/version-4.4/building-extensions/index.md

```diff
@@ -2,10 +2,51 @@
 sidebar_position: 5
 ---
 Build Extensions
-=======================
-This is the content for tutorials how to build extensions
-:::caution TODO
+================
+Joomla is a rich featured content management system, but if you're building a website with Joomla and you need extra features which aren't available by default, then you can easily extend it with extensions. There are five common types of extensions for Joomla: Components, Modules, Plugins, Templates, and Languages. There are three others: Packages, Files and Libraries. Each of these extensions handle specific functionality (many built-in features of Joomla are implemented using extensions).
 
-This page is unfinished, please use the **Edit this Page** link at the bottom of this page to help make it more useful.
+![Screenshot showing extension types](./_assets/screenshot-extension-types.jpg)
 
-:::
+# Components
+[Components](./components/index.md) provide the central part of a web page on a Joomla site; each site web page displays the output from one component. They can be thought of as mini applications. Most components have two parts: a site part and an administrator part. For example, `com_content` is the component which handles articles; on the site front-end `com_content` displays articles to website visitors and on the back-end `com_content` provides the functionality for administrators to edit articles. 
```

Comment:
For components, I would add their main role. Kind of: Their main role is to manage the Data on the site, provides an interface for data creatin/removing/editting etc. They also process the Request. 

Manage Articles, Categories (Content component), manage products and orders (Shoping cart component), manage images and media files (Gallery and media component).

Extending functionality: like provide backup solutions, handle special request.

Maybe with some better wording :smiley: